includes/acl/decompression/database/impl/database.impl.h

```diff
@@ -34,6 +34,12 @@
 
 #include <cstdint>
 
+#if __cplusplus >= 202002L
```

Comment:
I'm afraid that `__cplusplus` isn't always defined. It is optional with MSVC and it requires a compilation flag to be enabled: https://learn.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=msvc-170

For MSVC, we would have to test for `_MSVC_LANG` instead (you can use the RTM_*_COMPILER macros to detect the compiler). I would suggest using the C++11 value as a fallback in an `#if defined(..)` chain for msvc/gcc/clang/etc.

Otherwise, great fix, thanks! I'll see if I can add C++20 compilation targets to CI so that I can catch deprecated things like this.

Reply:
Don't knew that `__cplusplus` is not always defined for MS systems. 
The next time will be a bit busy, but I can take a look into the proposed additional solution, somewhere I should also have a Windows for testing that out.

Reply:
Don't knew that `__cplusplus` is not always defined for MS systems. 
The next time will be a bit busy, but I can take a look into the proposed additional solution, somewhere I should also have a Windows for testing that out.

Reply:
I finally got around to adding support to test this locally. I'll take your changes and clean them up, thanks!